registry/storage/driver/swift/swift.go

```diff
@@ -0,0 +1,561 @@
+func (d *driver) Stat(ctx context.Context, path string) (storagedriver.FileInfo, error) {
+	info, _, err := d.Conn.Object(d.Container, d.swiftPath(path))
+	if err != nil {
+		return nil, parseError(path, err)
+	}
+
+	fi := storagedriver.FileInfoFields{
+		Path:    path,
+		IsDir:   info.ContentType == directoryMimeType,
+		Size:    info.Bytes,
+		ModTime: info.LastModified,
+	}
+
+	return storagedriver.FileInfoInternal{FileInfoFields: fi}, nil
+}
+
+// List returns a list of the objects that are direct descendants of the given path.
+func (d *driver) List(ctx context.Context, path string) ([]string, error) {
+	var files []string
+
+	prefix := d.swiftPath(path)
+	if prefix != "" {
+		prefix += "/"
+	}
+
+	opts := &swift.ObjectsOpts{
+		Prefix:    prefix,
+		Delimiter: '/',
+	}
+
+	objects, err := d.Conn.Objects(d.Container, opts)
+	for _, obj := range objects {
+		if !obj.PseudoDirectory {
+			files = append(files, strings.TrimPrefix(strings.TrimSuffix(obj.Name, "/"), d.swiftPath("/")))
+		}
+	}
+
+	return files, parseError(path, err)
+}
+
+// Move moves an object stored at sourcePath to destPath, removing the original
+// object.
+func (d *driver) Move(ctx context.Context, sourcePath string, destPath string) error {
+	err := d.Conn.ObjectMove(d.Container, d.swiftPath(sourcePath),
+		d.Container, d.swiftPath(destPath))
+	if err != nil {
+		return parseError(sourcePath, err)
+	}
+
+	return nil
+}
+
+// Delete recursively deletes all objects stored at "path" and its subpaths.
+func (d *driver) Delete(ctx context.Context, path string) error {
+	opts := swift.ObjectsOpts{
+		Prefix: d.swiftPath(path),
+	}
+
+	objects, err := d.Conn.ObjectNamesAll(d.Container, &opts)
+	if err != nil {
+		return parseError(path, err)
+	}
+	if len(objects) == 0 {
+		return storagedriver.PathNotFoundError{Path: path}
+	}
+
+	if d.BulkDeleteSupport {
+		if _, err := d.Conn.BulkDelete(d.Container, objects); err != swift.Forbidden {
+			return parseError(path, err)
+		}
+	}
+
+	for _, name := range objects {
+		if _, headers, err := d.Conn.Object(d.Container, name); err == nil {
+			manifest, ok := headers["X-Object-Manifest"]
+			if ok {
+				components := strings.SplitN(manifest, "/", 2)
+				segContainer := components[0]
+				segments, err := d.getAllSegments(components[1])
+				if err != nil {
+					return parseError(name, err)
+				}
+
+				for _, s := range segments {
+					if err := d.Conn.ObjectDelete(segContainer, s.Name); err != nil {
+						return parseError(s.Name, err)
+					}
+				}
+			}
+		} else {
+			return parseError(name, err)
+		}
+
+		if err := d.Conn.ObjectDelete(d.Container, name); err != nil {
+			return parseError(name, err)
+		}
+	}
+
+	return nil
+}
+
+// URLFor returns a URL which may be used to retrieve the content stored at the given path.
+// May return an UnsupportedMethodErr in certain StorageDriver implementations.
+func (d *driver) URLFor(ctx context.Context, path string, options map[string]interface{}) (string, error) {
```

Comment:
Is there a plan to implement this? I believe the way to do it with openstack is with a Temporary URL?